src/service_router.rs

```diff
@@ -104,71 +104,77 @@ pub fn create<T: Controller>(controller: T) -> Chain {
-    }
+    describe! services {
+        before_each {
+            use iron::Headers;
+            use controller::FoxBox;
+            use iron_test::{ request, response };
+
+            let controller = FoxBox::new(false, Some("localhost".to_owned()), None, None);
+            let service_router = create(controller.clone());
+        }
 
-    it "should create list.json" {
-        let response = request::get("http://localhost:3000/list.json",
-                        Headers::new(),
-                        &service_router).unwrap();
+        it "should create list.json" {
+            let response = request::get("http://localhost:3000/list.json",
+                            Headers::new(),
+                            &service_router).unwrap();
 
-        let result = response::extract_body_to_string(response);
-        assert_eq!(result, "[]");
-    }
+            let result = response::extract_body_to_string(response);
+            assert_eq!(result, "[]");
+        }
 
-    it "should make service available" {
-        use controller::Controller;
-        use stubs::service::ServiceStub;
-        controller.add_service(Box::new(ServiceStub));
-        let response = request::get("http://localhost:3000/1/a-command",
-                        Headers::new(),
-                        &service_router).unwrap();
+        it "should make service available" {
+            use controller::Controller;
+            use stubs::service::ServiceStub;
+            controller.add_service(Box::new(ServiceStub));
+            let response = request::get("http://localhost:3000/1/a-command",
+                            Headers::new(),
+                            &service_router).unwrap();
 
-        let result = response::extract_body_to_string(response);
-        assert_eq!(result, "request processed");
-    }
+            let result = response::extract_body_to_string(response);
+            assert_eq!(result, "request processed");
+        }
 
-    it "should return an error if no service was found" {
-        let response = request::get("http://localhost:3000/unknown-id/a-command",
-                        Headers::new(),
-                        &service_router).unwrap();
+        it "should return an error if no service was found" {
+            let response = request::get("http://localhost:3000/unknown-id/a-command",
+                            Headers::new(),
+                            &service_router).unwrap();
 
-        let result = response::extract_body_to_string(response);
-        assert_eq!(result, "No Such Service: unknown-id");
+            let result = response::extract_body_to_string(response);
+            assert_eq!(result, "No Such Service: unknown-id");
+        }
     }
 
-    it "should get the appropriate CORS headers" {
-        use iron::headers;
-        use super::CORS;
+    describe! cors {
+        before_each {
+            use iron::Headers;
+            use controller::FoxBox;
+            use iron_test::request;
```

Comment:
Nit: Could we factorize these imports in a `before_each` in the `service_router` scope?